This was:

>From 9fd6bcfe09e4dbe607b3cc5f533f414b519890a0 Mon Sep 17 00:00:00 2001
From: Maxim Kuvyrkov <maxim@codesourcery.com>
Date: Tue, 4 May 2010 02:25:50 -0700
Subject: [PATCH 5/7] Fix gthr-posix.h to support Bionic

but has been rebased onto gcc-4.4.4

--- gcc-4.4.4/gcc/gthr-posix.h~	Fri Apr 10 09:23:07 2009
+++ gcc-4.4.4/gcc/gthr-posix.h	Thu Jun 24 18:49:12 2010
@@ -124,7 +124,9 @@
 __gthrw(pthread_equal)
 __gthrw(pthread_self)
 __gthrw(pthread_detach)
+#ifndef __BIONIC__
 __gthrw(pthread_cancel)
+#endif
 __gthrw(sched_yield)
 
 __gthrw(pthread_mutex_lock)
@@ -238,8 +240,16 @@
 __gthread_active_p (void)
 {
   static void *const __gthread_active_ptr 
-    = __extension__ (void *) &__gthrw_(pthread_cancel);
-  return __gthread_active_ptr != 0;
+    = __extension__ (void *) &__gthrw_(
+/* Android's C library does not provide pthread_cancel, check for
+   `pthread_create' instead.  */
+#ifndef __BIONIC__
+				       pthread_cancel
+#else
+				       pthread_create
+#endif
+				       );
+ return __gthread_active_ptr != 0;
 }
 
 #endif /* FreeBSD or Solaris */

